default.nix

```diff
@@ -1,64 +1,14 @@
-{ pkgs ? (import ./nix/nixpkgs.nix) }:
-let
-  source = [
-    "^.*\.md"
-    "^benchmark.*$"
-    "^examples.*$"
-    "^models.*$"
-    "^monad-bayes\.cabal$"
-    "^src.*$"
-    "^stack.*\.yaml$"
-    "^test.*$"
-  ];
-  mkMonadBayes = stackYaml: doBenchAndTest:
+(
+  import
+  (
     let
-      project = pkgs.haskell-nix.stackProject {
-        inherit stackYaml;
-        src = pkgs.lib.sourceByRegex ./. source;
-      };
-      # https://github.com/input-output-hk/haskell.nix/issues/470 suggests a
-      # better way to build benchmarks.
-      benchmarks = project.monad-bayes.components.benchmarks;
+      lock = builtins.fromJSON (builtins.readFile ./flake.lock);
     in
-    pkgs.recurseIntoAttrs
-      {
-        "shellFor" = project.shellFor;
-        "lib-and-test" = project.monad-bayes;
-      } // (
-      if doBenchAndTest
-      then {
-        "ssm-bench" = benchmarks.ssm-bench;
-        "speed-bench" = benchmarks.speed-bench;
-        "checks" = project.monad-bayes.checks;
+      fetchTarball {
+        url = "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
```

Comment:
Why not use `builtins.getFlake`? I know it's not available in older versions of nix, but it's much nicer than using this. Are there still a significant number of people using older nix versions? Do we have stats on that?

Reply:
Simple use `flake-compat` to provide compatibility with older version of Nix. This is also what is tested.

Adding `flake-compat` to `flake.nix` is just a sneaky way of using `flake.lock` to pin the `flake-compat` flake input.

Reply:
I understand the use. I'm just wondering if the copied/complicated code is warranted. Are there really that many nix users that haven't upgraded?

Reply:
Yes, there are many. By default on NixOS you still get Nix 2.3